File: 1910. Remove All Occurrences of a Substring/main.go
// total sam

package main

import "strings"

func removeOccurrences(s string, part string) string {
	stack := make([]string, 0)

	partRunes := []rune(part)

	checkStack := func() bool {
		if len(stack) < len(part) {
			return false
		}
		start := len(stack) - len(part)
		for i := start; i < len(stack); i++ {
			if stack[i] != string(partRunes[i-start]) {
				return false
			}
		}
		return true
	}

	for _, c := range s {
		stack = append(stack, string(c))
		if checkStack() {
			stack = stack[:len(stack)-len(part)]
		}
	}

	return strings.Join(stack, "")
}

File: 1910. Remove All Occurrences of a Substring/main.py
# total sam

class Solution:
    def removeOccurrences(self, s: str, part: str) -> str:
        stack = []

        def check_stack():
            if len(stack) < len(part):
                return False
            start = len(stack) - len(part)
            for i in range(start, len(stack)):
                if stack[i] != part[i - start]:
                    return False
            return True

        for c in s:
            stack.append(c)
            if not check_stack():
                continue
            for _ in range(len(part)):
                stack.pop()
        
        return "".join(stack)